controllers/storagecluster/reconcile.go

```diff
@@ -532,6 +532,17 @@ func (r *StorageClusterReconciler) reconcilePhases(
 		}
 	}
 
+	// For security of encrypted messages, When both encryption and compression are enabled,
+	// compression setting will be ignored and message will not be compressed.
+	// Refer https://docs.ceph.com/en/quincy/rados/configuration/msgr2/#confval-ms_compress_secure
+	networkSpec := instance.Spec.Network
+	if networkSpec != nil && networkSpec.Connections != nil &&
+		networkSpec.Connections.Encryption != nil && networkSpec.Connections.Encryption.Enabled &&
+		networkSpec.Connections.Compression != nil && networkSpec.Connections.Compression.Enabled {
+		r.Log.Info("Both in-transit encryption & compression are enabled. ")
+		r.Log.Info("To protect security of encrypted messages ceph will ignore the compression setting by default here.")
```

Comment:
Divide the line into 2 but do not use r.log twice. I would also like to have an k8s event for the same.